app/controllers/api/v1/passwords_controller.rb

```diff
@@ -2,16 +2,23 @@
 
 module API
   module V1
-    class PasswordsController < Devise::PasswordsController
-      skip_before_action :verify_authenticity_token
+    class PasswordsController < ApplicationController
+      before_action :verify_oauth_application
```

Comment:
@hoangmirs How about we keep devise implementation the same and keep inheriting from `Devise::PasswordsController`, for the `verify_oauth_application`, we can move it to module and include the module here.